### PR TITLE
Do not throw exception on PhpcrMenuProvider::has() method

### PR DESCRIPTION
**Pull request by ahmadrabie**
/cc @WouterJ @lsmith77

### CODE REVIEW

File: Provider/PhpcrMenuProvider.php [hunk 1]
```diff
@@ -188,7 +189,11 @@ public function get($name, array $options = array())
      */
     public function has($name, array $options = array())
     {
-        return $this->find($name, $options, false) instanceof NodeInterface;
+        try {
+            return $this->find($name, $options, false) instanceof NodeInterface;
+        } catch (NamespaceException $e) {
```
Comment (dbu) [inline Provider/PhpcrMenuProvider.php]: do you get a namespace exception when the name has a colon in it? that indeed should not break us. there could be other invalid paths i think, e.g. with a:b:c - can you try with that too and see what exception is thrown? i don't want to catch RepositoryException, as that would hide if the phpcr is down or fundamental setup mistakes.

we could also check whether $name is a valid phpcr name - that would avoid the complete roundtrip to phpcr if you have non-phpcr menues. this method from phpcr-utils could help: https://github.com/phpcr/phpcr-utils/blob/master/src/PHPCR/Util/PathHelper.php#L47

Comment (ahmadrabie) [inline Provider/PhpcrMenuProvider.php]: @dbu Yes I had colon in my name (It was actually a `callable` (!) like `AcmeBundle:Menu:mainMenu`) that I was passing to `knp_menu_get`. But it seems that CmfMenuProvider was breaking the chain with a `NamespaceException`.

I'm not a Cmf/Knp expert but maybe it's better to force a `@` before the name when you want to pass a callable.

/cc @stof

**Comment by dbu** [comment]
i think this really is a bug that we need to fix. the menu provider should not interfere with other systems.

i think the rebasing on master created some problems. but this is a bugfix and should be done against the 1.2 branch.

i will try a fix that handles more situations.

**Comment by dbu** [comment]
i created #225 against 1.2 branch. that would then need to be ported to 2.0 as well, once its merged.

File: Provider/PhpcrMenuProvider.php (hunk 1 above)
Comment (stof) [inline Provider/PhpcrMenuProvider.php]: any name is valid. So the has method is not allowed to throw any exception if it does not have this menu. If the naming does not match what it supports, it must return false to let another provider handle it

**Pull request closed without merging**